oioswift/common/middleware/copy.py

```diff
@@ -186,6 +187,54 @@ def _load_object_post_as_copy_conf(self, conf):
         except (NoSectionError, NoOptionError):
             pass
 
+    def allow_fast_copy(self, req):
+        """Check is fastcopy is possible."""
+
+        if req.method != 'PUT' or 'X-Copy-From' not in req.headers:
+            return False
+
+        # TODO maybe is it not necessary
+        already_checked = req.headers.get("Oio-Check-Fastcopy", 0)
+        if already_checked:
+            return False
+
+        self.logger.debug("COPY check fastcopy")
+
+        # mark request already checked
+        req.headers['Oio-Check-Fastcopy'] = 1
+
+        if req.headers.get('Range') or \
+                req.headers.get('X-Amz-Copy-Source-Range'):
+            self.logger.debug(
+                "COPY fastcopy not available (reason=Range header)")
+            return False
+
+        # now check if object is SLO:
+        # when a SLO is used as source without range read,
+        # it is used to recreate a plain object
+        # TODO we may cheat to create object as appended metachunks ?
+
+        req2 = req.environ.copy()
+        # FIXME check if a proper method is available
+        req2['PATH_INFO'] = '/'.join(req2['PATH_INFO'].split('/')[0:3]) + '/'
```

Comment:
`Request` objects have a `path` property, and you can use `swift.common.swob.split_path` to parse it (or more simply `req2.split_path()`).